lang/lua/lua.py

```diff
@@ -0,0 +1,312 @@
+from talon import Context, Module, actions, settings
+
+mod = Module()
+ctx = Context()
+ctx.matches = r"""
+tag: user.lua
+"""
+
+mod.setting(
+    "lua_version",
+    type=float,
+    default=5.1,
+    desc="The default lua version to use. Dictates certain operators",
+)
+mod.tag("stylua", desc="Tag for stylua linting commands")
+
+ctx.lists["user.code_common_function"] = {
+    "to number": "tonumber",
+    "eye pairs": "ipairs",
+    "print": "print",
+    "print F": "printf",
+    "type": "type",
+    "assert": "assert",
+    "get meta table": "getmetatable",
+    "set meta table": "setmetatable",
+    # io
+    "eye oh write": "io.write",
+    "eye oh read": "io.read",
+    "eye oh open": "io.open",
+    # string
+    "format": "string.format",
+    "string G find": "string.gfind",
+    "string find": "string.strfind",
+    "string len": "string.strlen",
+    "string upper": "string.strupper",
+    "string lower": "string.strlower",
+    "string sub": "string.strsub",
+    "string G sub": "string.gsub",
+    "string match": "string.match",
+    "string G match": "string.gmatch",
+    # table
+    "table unpack": "table.unpack",
+    "table insert": "table.insert",
+    "tabel get N": "table.getn",
+    "tabel sort": "table.sort",
+    # math
+    "math max": "math.max",
+    # json
+    "jason parse": "json.parse",
+    # http
+    "web get": "http.get",
+    # os
+    "O S date": "os.date",
+    "O S time": "os.time",
+    "O S clock": "os.clock",
+    "O S rename": "os.rename",
+    "O S remove": "os.remove",
+    "O S getenv": "os.getenv",
+    "O S execute": "os.execute",
+}
+
+ctx.lists["user.code_libraries"] = {
+    "bit": "bit",
+    "I O": "io",
+    "string": "string",
+    "U T F eight": "utf8",
+    "table": "table",
+    "math": "math",
+    "O S": "os",
+    "debug": "debug",
+    "L F S": "lfs",
+    "socket": "socket",
+    "H T T P": "http",
+    "web": "http",
+    "jason": "json",
+}
+
+
+@mod.capture(rule="{self.lua_functions}")
+def lua_functions(m) -> str:
+    "Returns a string"
+    return m.lua_functions
+
+
+@ctx.action_class("user")
+class UserActions:
+    # tag-related actions listed first, indicated by comment. corresponds to
+    # the tag(): user.code_imperative style declaration in the language .talon
+    # file
+
+    ##
+    # code_imperative
+    ##
+    def code_state_if():
+        actions.insert("if  then\n\nend")
```

Comment:
See comment here about not including end block delimiters: https://github.com/knausj85/knausj_talon/pull/1162#discussion_r1160410725 (let's discuss there if needed). This looks like it applies to all your control blocks and public/private function actions.